src/routes/safe/components/Transactions/TxList/utils.ts

```diff
@@ -136,6 +157,91 @@ export const getTxTo = ({ txInfo }: Pick<Transaction, 'txInfo'>): AddressEx | un
   }
 }
 
+export const getTxInfo = (transaction: Transaction, safeAddress: string): TxInfoProps => {
+  if (!transaction.txDetails) return {} as TxInfoProps
+
+  const data = transaction.txDetails.txData?.hexData ?? EMPTY_DATA
+  const baseGas = isMultiSigExecutionDetails(transaction.txDetails.detailedExecutionInfo)
+    ? transaction.txDetails.detailedExecutionInfo.baseGas
+    : '0'
+  const gasPrice = isMultiSigExecutionDetails(transaction.txDetails.detailedExecutionInfo)
+    ? transaction.txDetails.detailedExecutionInfo.gasPrice
+    : '0'
+  const safeTxGas = isMultiSigExecutionDetails(transaction.txDetails.detailedExecutionInfo)
+    ? transaction.txDetails.detailedExecutionInfo.safeTxGas
+    : '0'
+  const gasToken = isMultiSigExecutionDetails(transaction.txDetails.detailedExecutionInfo)
+    ? transaction.txDetails.detailedExecutionInfo.gasToken
+    : ZERO_ADDRESS
+  const nonce = (transaction.executionInfo as MultisigExecutionInfo)?.nonce ?? 0
+  const refundReceiver = isMultiSigExecutionDetails(transaction.txDetails.detailedExecutionInfo)
+    ? transaction.txDetails.detailedExecutionInfo.refundReceiver.value
+    : ZERO_ADDRESS
```

Comment:
I'd suggest either an early return with the 'template' transaction or reassigning them all in one go if `isMultiSigExecutionDetails()`.